Examples/ObjectiveC/ClusteringExample.m

```diff
@@ -39,37 +39,33 @@ - (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+                             @100: [UIColor redColor],
+                             @200: [UIColor purpleColor] };
     // Show clustered features as circles. The `point_count` attribute is built into clustering-enabled source features.
     MGLCircleStyleLayer *circlesLayer = [[MGLCircleStyleLayer alloc] initWithIdentifier:@"clusteredPorts" source:source];
-    circlesLayer.circleRadius = [MGLStyleValue valueWithRawValue:@(self.icon.size.width / 2)];
-    circlesLayer.circleOpacity = [MGLStyleValue valueWithRawValue:@0.75];
-    circlesLayer.circleStrokeColor = [MGLStyleValue valueWithRawValue:[[UIColor whiteColor] colorWithAlphaComponent:0.75]];
-    circlesLayer.circleStrokeWidth = [MGLStyleValue valueWithRawValue:@2];
-    circlesLayer.circleColor = [MGLSourceStyleFunction
-        functionWithInterpolationMode:MGLInterpolationModeInterval
-                                stops:stops
-                        attributeName:@"point_count"
-                              options:nil];
+    circlesLayer.circleRadius = [NSExpression expressionForConstantValue:@(self.icon.size.width / 2)];
+    circlesLayer.circleOpacity = [NSExpression expressionForConstantValue:@0.75];
+    circlesLayer.circleStrokeColor = [NSExpression expressionForConstantValue:[[UIColor whiteColor] colorWithAlphaComponent:0.75]];
+    circlesLayer.circleStrokeWidth = [NSExpression expressionForConstantValue:@2];
+    circlesLayer.circleColor = [NSExpression expressionWithFormat:@"mgl_step:from:stops:(point_count, %@, %@)",
+                                [UIColor lightGrayColor], stops];
     circlesLayer.predicate = [NSPredicate predicateWithFormat:@"%K == YES", @"cluster"];
     [style addLayer:circlesLayer];
 
-    // Label cluster circles with a layer of text indicating feature count. Per text token convention, wrap the attribute in {}.
+    // Label cluster circles with a layer of text indicating feature count. The value for `point_count` is an integer. In order to use that value for the `MGLSymbolStyleLayer.text` property, cast it as a string. 
     MGLSymbolStyleLayer *numbersLayer = [[MGLSymbolStyleLayer alloc] initWithIdentifier:@"clusteredPortsNumbers" source:source];
-    numbersLayer.textColor = [MGLStyleValue valueWithRawValue:[UIColor whiteColor]];
-    numbersLayer.textFontSize = [MGLStyleValue valueWithRawValue:@(self.icon.size.width / 2)];
-    numbersLayer.iconAllowsOverlap = [MGLStyleValue valueWithRawValue:@(YES)];
-    numbersLayer.text = [MGLStyleValue valueWithRawValue:@"{point_count}"];
+    numbersLayer.textColor = [NSExpression expressionForConstantValue:[UIColor whiteColor]];
+    numbersLayer.textFontSize = [NSExpression expressionForConstantValue:@(self.icon.size.width / 2)];
+    numbersLayer.iconAllowsOverlap = [NSExpression expressionForConstantValue:@(YES)];
+    numbersLayer.text = [NSExpression expressionWithFormat:@"CAST(point_count, 'NSString')"];
     numbersLayer.predicate = [NSPredicate predicateWithFormat:@"%K == YES", @"cluster"];
```

Comment:
This format string doesn’t need a placeholder. `cluster == YES` is easier to read.